service/src/main/java/eu/europa/ec/fisheries/uvms/movement/service/bean/MovementFiltererBean.java

```diff
@@ -0,0 +1,79 @@
+Free Software Foundation, either version 3 of the License, or any later version. The IFDM Suite is distributed in
+the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details. You should have received a
+copy of the GNU General Public License along with the IFDM Suite. If not, see <http://www.gnu.org/licenses/>.
+*/
+package eu.europa.ec.fisheries.uvms.movement.service.bean;
+
+import eu.europa.ec.fisheries.schema.movement.area.v1.AreaType;
+import eu.europa.ec.fisheries.uvms.movement.model.exception.InvalidArgumentException;
+import eu.europa.ec.fisheries.uvms.movement.model.exception.MovementModelRuntimeException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class MovementFiltererBean {
+
+    private final static Logger LOG = LoggerFactory.getLogger(MovementFiltererBean.class);
+
+    @Inject
+    private MovementService movementService;
+
+    public List<String> filterGuidListForPeriodAndAreaTypesByArea(List<String> guidList, Date startDate, Date endDate, List<AreaType> areaTypes) throws MovementModelRuntimeException {
+
+        if(guidList == null || guidList.isEmpty()) throw new InvalidArgumentException("No Guid list to filter");
+        if(startDate == null) throw new InvalidArgumentException("No start date provided/or invalid syntax, try UTC");
+        if(endDate == null) throw new InvalidArgumentException("No end date provided/or invalid syntax, try UTC");
+        if(startDate.toInstant().isAfter(endDate.toInstant())) throw new InvalidArgumentException("Start date cannot be after end date");
+        if(areaTypes == null || areaTypes.isEmpty()) throw new InvalidArgumentException("Area type list was empty");
+
+        return guidList.stream()
+                .filter(new MovementFilterer(startDate,endDate,areaTypes)::filter)
+                .collect(Collectors.toList());
+    }
+
+    private class MovementFilterer{
+
+        List<Long> movementAreaIds;
+        List<AreaType> areaTypes;
+        Date startDate,endDate;
+
+        MovementFilterer(Date startDate, Date endDate,List<AreaType> areaTypes){
+            this.startDate = startDate;this.endDate = endDate;this.areaTypes = areaTypes;
+
+            movementAreaIds  = movementService.findMovementAreaIdsByAreaRemoteIdAndNameList(
+                    areaTypes.stream().map(AreaType::getAreaId)
+                            .map(String::valueOf)
+                            .collect(Collectors.toList()),
+                    areaTypes.stream().map(AreaType::getAreaName)
+                            .collect(Collectors.toList()));
+        }
+
+        boolean filter(String connectId){
+            if(connectId.length() != 36){
```

Comment:
Is this really necessary? Is there a good chance to get something that is not a UUID here? IF not, then remove this, it is confusing!